### DISCUSSION TITLE
Strange corners when using hyprland's layerrules for blur

### DISCUSSION THREAD

**Discussion by psi4j** [H3rmt/hyprshell#85] [Q&A] [answered]
I have added transparency in the background for hyprswitch's CSS like such:
```
window {
  --border-color: rgba(90, 90, 110, 0.4);
  --border-color-active: rgba(239, 9, 9, 0.9);
  --bg-color: rgba(0, 0, 0, 0.5);
  --bg-color-hover: rgba(10, 10, 10, 0.5);
  --index-border-color: rgba(84, 81, 106, 9);
  --border-radius: 11px;
  background-color: var(--bg-color);
  clip-path: inset(0 round var(--border-radius)); /* Clip blur effect */
}
```
I tried using clip-path to trim the sharp corners of the blur effect created by hyprland's layerrule set using:
```
layerrule = blur, gtk4-layer-shell

```
When ever I run hyprswitch with these settings, it creates these small corners, even though I've matched the rounding of the corners to 11px on both hyprland.conf and in hyprswitch's .CSS file. I don't know if this is a hyprswitch issue, so I just wanted to know if anyone has any advice as to how to fix this. Here's a shot of what I am referring to:
![image](https://github.com/user-attachments/assets/f82facd7-c435-45dd-96c3-69faea341abf)

I'd be really grateful for any advice you guys might have on both the Hyprland and hyprswitch side of things.

**Comment by H3rmt** [comment]
you could try to add the styles to `.monitor` and see if it changes anything.

The window isnt rounded, only the `.monitor` box is.

**Comment by psi4j** [comment, marked as answer]
I found out how to fix this. In hyprland.conf, you just add this line:
```hyprlang
layerrule = ignorezero, hyprswitch # <-----
layerrule = blur, hyprswitch # You can add background blur with this line too
```
